### ISSUE TITLE
2 further roles needed

### ISSUE THREAD

**Issue by dj4oC** [owncloud/ocis#10747]
Reference:
https://github.com/owncloud/ocis/issues/9698

The ticket empowers the user to activate / deactivate to share file versions. Unfortunately, this only works with Project Spaces. It is necessary to support that as well for the personal space.

Please add the following two roles:
“FileEditorListGrants**With**Versions” - Permission set: View, download, edit and show all invited people, **show all versions** - Type: File

“EditorListGrants**With**Versions” - Permission set: View, download, upload, edit, delete and show all invited people, **show all versions**. - Type: Folder

### New permissions table would look like this:

Name | Id | FILE | FOLDER | SPACE | Default | Permissions | Permissions | Permissions | Permissions | Permissions | Permissions | Permissions | Permissions | Permissions
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
Viewer | b1e2218d-eef8-4d4c-b82d-0f1a1b48f3b5 | YES | YES |   | YES | View | download |   |   |   |   |   |   |  
Editor | fb6c3e19-e378-47e5-b277-9732f9de6e21 |   | YES |   | YES |   | download | upload | edit | delete |   |   |   |  
FileEditor | 2d00ce52-1fc2-4dbc-8b95-a73b73395f5a | YES |   |   | YES | View | download |   | edit |   |   |   |   |  
FolderEditorLite | 1c996275-f1c9-4e71-abdf-a42f6495e960 |   | YES |   | YES | View | download | upload |   |   |   |   |   |  
ViewerListGrants | d5041006-ebb3-4b4a-b6a4-7c180ecfb17d | YES | YES |   | NO | View | download |   |   |   | show all invited people |   |   |  
EditorListGrants | e8ea8b21-abd4-45d2-b893-8d1546378e9e |   | YES |   | NO | View | download | upload | edit | delete | show all invited people |   |   |  
**EditorListGrantsWithVersions** |   |   | YES |   | NO | View | download | upload | edit | delete | show all invited people | **show history**  |   |  
FileEditorListGrants | c1235aea-d106-42db-8458-7d5610fb0a67 | YES |   |   | NO | View | download |   | edit |   | show all invited people |   |   |  
**FileEditorListGrantsWithVersions** |   | YES |   |   | NO | View | download |   | edit |   | show all invited people | **show history** |   |  
SecureViewer | aa97fe03-7980-45ac-9e50-b325749fd7e6 | YES | YES |   | NO |   |   |   |   |   |   |   |   | View only documents, images and PDFs. Watermarks   will be applied.
SpaceViewer | a8d5fe5e-96e3-418d-825b-534dbdf22b99 |   |   | YES | YES | View | download |   |   |   |   |   |   |  
SpaceEditor | 58c63c02-1d89-4572-916a-870abc5a1b7d |   |   | YES | YES | View | download | upload | edit | delete |   | show history |   |  
SpaceEditorWithoutVersions | 3284f2d5-0070-4ad8-ac40-c247f7c1fb27 |   |   | YES | NO | View | download | upload | edit | delete |   |   |   |  
RoleManager | 312c0871-5ef7-4b3a-85b6-0e4074c64049 |   |   | YES | YES | View | download | upload | edit | delete |   |   | manage members |

**Comment by kobergj** [comment]
@dj4oC  I dont quite understand the issue. Please keep in mind that sharing roles (= share something else than a spaceroot) are completely different from space roles (= roles you have on a space root).

Sharing roles are the same on personal and project spaces.

**Comment by dj4oC** [comment]
Yes. Right now the decision is done on space level. The request is to do it on file and folder level

**Comment by micbar** [comment]
@dj4oC i also do not understand the problem.

Would be better to formulate the problem before suggesting a possible solution.

**Comment by dj4oC** [comment]
@kobergj please let's talk about this issue

**Comment by kobergj** [comment]
@dj4oC tell me first: does it need to be in `7.1` release?

**Comment by dj4oC** [comment]
It better is

**Comment by bastianbeier** [comment]
For showing the changes in the UI it seems that there are changes needed in the web @kobergj @LukasHirt @mzner 

This request seems to be missing:
`curl --location --request PROPFIND 'https://localhost:9200/remote.php/dav/meta/${SHARE_ID}/v' \
--data '<d:propfind xmlns:d="DAV:" xmlns:oc="http://owncloud.org/ns">
  <d:prop/>
</d:propfind>`
